pkg/storage/unified/search/bleve_mappings.go

```diff
@@ -1,19 +1,55 @@
 package search
 
 import (
+	"log"
+
 	"github.com/blevesearch/bleve/v2"
+	"github.com/blevesearch/bleve/v2/analysis/analyzer/custom"
 	"github.com/blevesearch/bleve/v2/analysis/analyzer/keyword"
+	"github.com/blevesearch/bleve/v2/analysis/token/ngram"
+	"github.com/blevesearch/bleve/v2/analysis/tokenizer/unicode"
 	"github.com/blevesearch/bleve/v2/mapping"
 
 	"github.com/grafana/grafana/pkg/storage/unified/resource"
 )
 
 func getBleveMappings(fields resource.SearchableDocumentFields) mapping.IndexMapping {
 	mapper := bleve.NewIndexMapping()
+	// this will tokenize substrings for ngram search
+	// Example: "hello" -> "hel", "ell", "llo"
+	// this will allow us to search for a substring of a word without using wildcards which are slow and consumer a lot of memory
+	addNgramCustomAnalyzer(mapper)
 	mapper.DefaultMapping = getBleveDocMappings(fields)
+
 	return mapper
 }
 
+func addNgramCustomAnalyzer(mapper *mapping.IndexMappingImpl) {
+	// Define an N-Gram tokenizer (for substring search)
```

Comment:
Check: Maybe worth checking https://github.com/grafana/grafana/blob/main/pkg/services/searchV2/ngram.go#L30

for that we did 0-7 with ngrams, and then change the query based on length:
https://github.com/grafana/grafana/blob/main/pkg/services/searchV2/bluge.go#L464